### PR TITLE
Service Incident Rule Resource Addition

### PR DESCRIPTION
**Pull request by msg4sunny**

### CODE REVIEW

File: opsgenie/resource_opsgenie_service_incident_rule.go [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+	return &schema.Resource{
+		Create: resourceOpsGenieServiceIncidentRuleCreate,
+		Read:   handleNonExistentResource(resourceOpsGenieServiceIncidentRuleRead),
+		Update: resourceOpsGenieServiceIncidentRuleUpdate,
+		Delete: resourceOpsGenieServiceIncidentRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idParts := strings.Split(d.Id(), "/")
+				if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+					return nil, fmt.Errorf("Unexpected format of ID (%q), expected team_id/notification_policy_id", d.Id())
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_service_incident_rule.go]: notif_policy -> service_id

File: opsgenie/resource_opsgenie_service_incident_rule.go [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+	if err != nil {
+		return err
+	}
+
+	service_id := d.Get("service_id").(string)
+	createRequest := &service.CreateIncidentRuleRequest{
+		ServiceId: service_id,
+	}
+
+	//if len(d.Get("conditions").([]interface{})) > 0 {
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_service_incident_rule.go]: unnecessary comment

File: opsgenie/resource_opsgenie_service_incident_rule.go [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+	incident_rule := d.Get("incident_rule").([]interface{})
+	for _, v := range incident_rule {
+		config := v.(map[string]interface{})
+		createRequest.ConditionMatchType = og.ConditionMatchType(config["condition_match_type"].(string))
+		createRequest.Conditions = expandOpsGenieServiceIncidentRuleConditions(config["conditions"].([]interface{}))
+		createRequest.IncidentProperties = expandOpsGenieServiceIncidentRuleIncidentProperties(config["incident_properties"].([]interface{}))
+	}
+	//}
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_service_incident_rule.go]: same

File: opsgenie/resource_opsgenie_service_incident_rule.go [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+
+	service_id := d.Get("service_id").(string)
+	incident_rule_id := d.Id()
+
+	updateRequest := &service.UpdateIncidentRuleRequest{
+		ServiceId:      service_id,
+		IncidentRuleId: incident_rule_id,
+	}
+
+	//if len(d.Get("conditions").([]interface{})) > 0 || len(d.Get("incident_properties").([]interface{})) > 0 {
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_service_incident_rule.go]: do we need this ?

File: opsgenie/resource_opsgenie_service_incident_rule.go [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+		stakeholder_properties.Enable = &enable
+		stakeholder_properties.Message = config["message"].(string)
+		stakeholder_properties.Description = config["description"].(string)
+	}
+
+	return stakeholder_properties
+}
+
+func flattenOpsgenieServiceIncidentRuleRequestTags(input *schema.Set) []string {
+	//input := input_map.Get("tags").(*schema.Set)
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_service_incident_rule.go]: same

File: website/docs/r/service_incident_rule.html.markdown [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+* `description` - (Optional) Description that is generally used to provide a detailed information about the alert.
+
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - The ID of the Opsgenie Service Incident Policy.
+
```
Comment (ffahri) [inline website/docs/r/service_incident_rule.html.markdown:110]: dont we need any import line ?

**Review by ffahri — CHANGES_REQUESTED** [review]

File: opsgenie/resource_opsgenie_service_incident_rule_test.go [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+			return err
+		}
+		incident_rule_res, err := client.GetIncidentRules(context.Background(), &service.GetIncidentRulesRequest{
+			ServiceId: service_id,
+		})
+
+		if err != nil {
+			return fmt.Errorf("Bad: Service ID %q does not exist", service_id)
+		} else {
+			fmt.Println("SUNNY Error is NIL, moving forward")
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_service_incident_rule_test.go]: also this line should be generic of course

File: opsgenie/resource_opsgenie_service_incident_rule.go (hunk 1 above)
Comment (msg4sunny) [reply opsgenie/resource_opsgenie_service_incident_rule.go]: Done

File: opsgenie/resource_opsgenie_service_incident_rule.go (hunk 2 above)
Comment (msg4sunny) [reply opsgenie/resource_opsgenie_service_incident_rule.go]: Removed

File: opsgenie/resource_opsgenie_service_incident_rule_test.go (hunk 7 above)
Comment (msg4sunny) [reply opsgenie/resource_opsgenie_service_incident_rule_test.go]: Changed

File: website/docs/r/service_incident_rule.html.markdown (hunk 6 above)
Comment (msg4sunny) [reply website/docs/r/service_incident_rule.html.markdown:110]: Added

File: opsgenie/resource_opsgenie_service_incident_rule.go (hunk 5 above)
Comment (msg4sunny) [reply opsgenie/resource_opsgenie_service_incident_rule.go]: Removed

File: opsgenie/resource_opsgenie_service_incident_rule.go (hunk 4 above)
Comment (msg4sunny) [reply opsgenie/resource_opsgenie_service_incident_rule.go]: incident_properties is a mandatory field. So, No need to check this

File: opsgenie/resource_opsgenie_service_incident_rule.go (hunk 3 above)
Comment (msg4sunny) [reply opsgenie/resource_opsgenie_service_incident_rule.go]: done

**Review by ffahri — APPROVED** [review]

**Pull request merged**